docs/modules/traits/pages/builder.adoc

```diff
@@ -98,6 +98,33 @@ Syntax: [configmap\|secret]:name[/key], where name represents the resource name,
 | []string
 | A list of limit memory configuration for the specific task with format `<task-name>:<limit-memory-conf>`.
 
+| builder.node-selector
+| map[string]string
+| Defines a set of nodes the builder pod is eligible to be scheduled on, based on labels on the node.
+
 |===
 
 // End of autogenerated code - DO NOT EDIT! (configuration)
+
+== Node Selectors
+
+With this trait you will also be able to define node selectors for the `builder` pod when using the `pod`` build strategy.
+Here is a simple example:
+
+.template.yaml
+[source,yaml]
+----
+  traits:
+    builder:
+      nodeSelector:
+        size: large
+----
+
+The `builder` pod will be created with a node selector that allows it to run only on nodes where the `size` label is equal to `large`.
+
+Node selectors can be specified when running an integration with the CLI:
+
+[source,console]
+----
+$ kamel run --trait builder.node-selector.size=large integration.groovy
```

Comment:
I think this is something like `builder.node-selector.'size'=large`, is it?

Reply:
I tested it exactly like this and it worked. I think quotes are not required as there are spaces in the label.

Reply:
yeah, possibly in this case it works because "size" is a simple string. However, it may fail when using more complex annotations, so, better to be safe and keep the notation we're using for a map in the rest of the project.